app/helpers/languages_helper.rb

```diff
@@ -230,6 +230,34 @@ module LanguagesHelper
     'sr-Latn': 'Srpski (latinica)',
   }.freeze
 
+  # Helper for self.sorted_locale_keys
+  private_class_method def self.locale_name_for_sorting(locale)
+    if locale.blank? || locale == 'und'
+      '000'
+    elsif (supported_locale = SUPPORTED_LOCALES[locale.to_sym])
+      if /.*[a-zA-Z].*/.match?(supported_locale[1])
+        # Strip accents from Latin characters so e.g. Íslenska won't be bumped to the bottom.
+        # We have to exclude non-Latin languages from the transliterator, because results are dire.
+        ActiveSupport::Inflector.transliterate(supported_locale[1]).downcase
+      else
+        supported_locale[1].downcase
+      end
```

Comment:
Would using our `ASCIIFolding` module work better here?

Reply:
Thanks, I'll look into it! Less hacky is always better.

Reply:
Definitely much better! It also solves the problem of `ꆈꌠ꒿ Nuosuhxop` being on top.

BTW I spotted some duplicate languages in the list. I think we should rework the data structures to not categorize according to ISO standard, but according to the locales' scope within Mastodon. This will make the lists easier to understand & maintain.

In order to do that, we'll should merge all branches for locale changes/additions first, otherwise I'll get  merge conflicts. The PRs are:

- https://github.com/mastodon/mastodon/pull/26634 (already approved)
- https://github.com/mastodon/mastodon/pull/26015
- https://github.com/mastodon/mastodon/pull/27117
- https://github.com/mastodon/mastodon/pull/27136
- https://github.com/mastodon/mastodon/pull/27115